docs/datasets/data-anonymization/index.md

```diff
@@ -0,0 +1,221 @@
+  output_save_compressed_image: True # Save images as compressed images (True or False)
+  output_storage_id: "sqlite3" # Storage id for the output bag file (`sqlite3` or `mcap`)
+
+yolo:
+  model: "path/to/yolo/model" # Path to the trained YOLOv8 model file (`.pt` extension) (you can download the pre-trained model from releases)
+  config_path: "path/to/input/data.yaml" # Path to the config file of the dataset, which is created in the previous step
+  confidence: 0.15 # Confidence threshold for the YOLOv8 model (float)
+
+blur:
+  kernel_size: 31 # Kernel size for the Gaussian blur (int)
+  sigma_x: 11 # Sigma x for the Gaussian blur (int)
+```
+
+## Usage
+
+The tool provides two options to anonymize images in ROS 2 bag files.
+
+### Option 1: Anonymize with Unified Model
+
+You should provide a single rosbag and tool anonymize images in rosbag with a unified model. The model is a combination
+of GroundingDINO, OpenCLIP, YOLOv8 and SegmentAnything. If you don't want to use pre-trained YOLOv8 model, you can
+follow the instructions in the second option to train your own YOLOv8 model.
+
+You should set your configuration in config/anonymize_with_unified_model.yaml file.
+
+```bash
+python3 main.py config/anonymize_with_unified_model.yaml --anonymize_with_unified_model
+```
+
+### Option 2: Anonymize Using the YOLOv8 Model Trained on a Dataset Created with the Unified Model
+
+#### Step 1: Create a Dataset
+
+Create an initial dataset with the unified model. You can provide multiple ROS 2 bag files to create a dataset. After
+running the following command, the tool will create a dataset in YOLO format.
+
+You should set your configuration in config/yolo_create_dataset.yaml file.
+
+```bash
+python3 main.py config/yolo_create_dataset.yaml --yolo_create_dataset
+```
+
+#### Step 2: Manually Label the Missing Labels
+
+The dataset which is created in the first step has some missing labels. You should label the missing labels manually.
+
+#### Step 3: Split the Dataset
+
+Split the dataset into training and validation sets. Give the path to the dataset folder which is created in the first
+step.
+
+```bash
+autoware-rosbag2-anonymizer-split-dataset /path/to/dataset/folder
+```
+
+#### Step 4: Train the YOLOv8 Model
+
+Train the YOLOv8 model with the dataset which is created in the first step.
+
+You should set your configuration in config/yolo_train.yaml file.
+
+```bash
+python3 main.py config/yolo_train.yaml --yolo_train
+```
+
+#### Step 5: Anonymize Images in ROS 2 Bag Files
+
+Anonymize images in ROS 2 bag files with the trained YOLOv8 model. If you want to anonymize your ROS 2 bag file with only
+YOLOv8 model, you should use following command. But we recommend to use the unified model for better results. You can
+follow the Option 1 for the unified model with the YOLOv8 model trained by you.
+
+You should set your configuration in config/yolo_anonymize.yaml file.
+
+```bash
+python3 main.py config/yolo_anonymize.yaml --yolo_anonymize
```

Comment:
Some debug comments left on the 'autoware_rosbag2_anonymizer/tools/yolo_anonymize.py'.

Reply:
```
0: 416x640 2 license plates, 151.4ms
Speed: 3.7ms preprocess, 151.4ms inference, 132.2ms postprocess per image at shape (1, 3, 640, 640)
```

If you mentioned about these outputs, these comes from YOLO and I thought these looks nice.

Reply:
No, I mentioned about [these](https://github.com/autowarefoundation/autoware_rosbag2_anonymizer/blob/a87dd7a2cf59fb1883205a9203dc8607dd02e5a4/autoware_rosbag2_anonymizer/tools/yolo_anonymize.py#L65) debug comments. Also there are debug comments left on some other [[1]](https://github.com/autowarefoundation/autoware_rosbag2_anonymizer/blob/a87dd7a2cf59fb1883205a9203dc8607dd02e5a4/autoware_rosbag2_anonymizer/tools/anonymize_with_unified_model.py#L73) [[2]](https://github.com/autowarefoundation/autoware_rosbag2_anonymizer/blob/a87dd7a2cf59fb1883205a9203dc8607dd02e5a4/autoware_rosbag2_anonymizer/tools/yolo_create_dataset.py#L84C1-L97C29) files as well.

Reply:
I have some future plan on these debug parts which are comment right now, I think it can be stay for now.